src/consumption.cpp

```diff
@@ -161,6 +219,13 @@ edible_rating player::can_eat( const item &food, bool interactive, bool force )
         return INEDIBLE;
     }
 
+    for( const auto &m : food.type->materials ) {
+        if( !m.obj().edible() ) {
```

Comment:
Materials defining edibility is a giant hack. They shouldn't even define the "state of being a comestible", which is what should be forbidding eating.